src/dataregistry/DataRegistry.py

```diff
@@ -45,53 +49,84 @@ def __init__(
             Root directory for datasets, if None, default is assumed.
         verbose : bool
             True for more output.
+        site : str
+            Can be used instead of `root_dir`. Some predefined "sites" are
+            built in, such as "nersc", which will set the `root_dir` to the
+            data registry's default data location at NERSC.
         """
 
-        # Work out the location of the root directory
-        root_dir = self._get_root_dir(root_dir)
-
         # Establish connection to database
         self.db_connection = DbConnection(config_file, schema=schema, verbose=verbose)
 
-        # Create query object
-        self.Query = Query(self.db_connection)
+        # Work out the location of the root directory
+        root_dir = self._get_root_dir(root_dir, site)
 
         # Create registrar object
         self.Registrar = Registrar(
-            db_connection,
+            self.db_connection,
             root_dir,
             owner=owner,
             owner_type=owner_type,
         )
 
         # Create query object
-        self.Query = Query(db_connection, root_dir)
+        self.Query = Query(self.db_connection, root_dir)
 
-    def _get_root_dir(self, root_dir):
+    def _get_root_dir(self, root_dir, site):
         """
         What is the location of the root_dir we are pairing with?
 
-        Either this is a predefined "site", such as "NERSC", or a manually
-        passed path to the desired root directory. If `root_dir` is None then
-        "NERSC" is used as the default.
+        In order of priority:
+            - If manually passed `root_dir` is not None, use that.
+            - If manually passed `site` is not None, use that.
+            - If env DATAREG_SITE is set, use that.
+            - Else use `site="nersc"`.
+
+        All `site`s are assumed to be postgres. Sqlite users must manually
+        specify the `root_dir.
 
         Parameters
         ----------
         root_dir : str
+        site : str
 
         Returns
         -------
         - : str
             Path to root directory
         """
 
+        # Sqlite cannot work with `site`s
+        if self.db_connection._dialect == "sqlite":
+            if site is not None:
+                raise ValueError("Site's not allowed with Sqlite")
+
         # Load the site config yaml file
         with open(_SITE_CONFIG_PATH) as f:
             data = yaml.safe_load(f)
 
+        # Find out what the root_dir should be
         if root_dir is None:
-            return data["nersc"]
-        elif root_dir.lower() in data.keys():
-            return data[root_dir.lower()]
-        else:
-            return root_dir
+            if site is not None:
+                if site.lower() not in data.keys():
+                    raise ValueError(f"{site} is not a valid site")
+                root_dir = data[site.lower()]
+            elif os.getenv("DATAREG_SITE"):
```

Comment:
I don't think this section is quite right as it stands.   It should not be possible to use DATAREG_SITE with sqlite. In the end that has the same effect as specifying site directly, which you have rightly excluded above.   
For sqlite, root_dir must not be None. And it can't be one of the default site values, which you also have excluded.
It might be easier to read this section if all sqlite stuff comes first:

`with open(_SITE_CONFIG_PATH) as f:`
`    data = yaml.safe_load(f)`
`if self.db_connection._dialect == "sqlite":`
`    if root_dir is None:`
`        raise ValueError ...`
`    else:`
`          (code to check root_dir is not among default site values)`
`    return root_dir`

Then handle non-sqlite case

Reply:
Ah, I had in my head for some reason `DATAREG_SITE` was specifying `root_dir`. I've changed it as you suggest